common/cpp/react/renderer/components/rnscreens/RNSModalScreenComponentDescriptor.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <react/debug/react_native_assert.h>
+#include "RNSModalScreenShadowNode.h"
+#include <react/renderer/core/ConcreteComponentDescriptor.h>
+
+namespace facebook {
+namespace react {
+
+class RNSModalScreenComponentDescriptor final
+    : public ConcreteComponentDescriptor<RNSModalScreenShadowNode> {
+ public:
+  using ConcreteComponentDescriptor::ConcreteComponentDescriptor;
+
+  void adopt(ShadowNode& shadowNode) const override {
+    react_native_assert(
+        dynamic_cast<RNSModalScreenShadowNode*>(&shadowNode));
+    auto& screenShadowNode =
+        static_cast<RNSModalScreenShadowNode&>(shadowNode);
```

Comment:
I see that here we're asserting that shadowNode should be of type `RNSModalScreenShadowNode` (with the use of dynamic_cast), but then we're setting screenShadowNode as a static_cast. While this might work as usual, do you know if there are some edge cases when this might break? Or it won't work, when we will dynamic cast this shadow node as a `RNSModalScreenShadowNode`?

Reply:
I have no knowledge of it, I just copied it from `RNSScreenShadowNode` so it should work fine.

Reply:
AFAIK `dynamic_cast` and `static_cast` do the same thing but `dynamic_cast` does runtime type checks, while `static_cast` assumes that you know what you're doing. `react_native_assert` does nothing in release builds, so we do both casts only in debug.
If something is wrong, assert and `dynamic_cast` combination will catch that, but doing `static_cast` in production builds gets rid of type-safety overhead.